core/arch/arm/include/mm/core_mmu_arch.h

```diff
@@ -178,7 +178,17 @@ void map_memarea_sections(const struct tee_mmap_region *mm, uint32_t *ttb);
 static inline bool core_mmu_check_max_pa(paddr_t pa __maybe_unused)
 {
 #if defined(ARM64)
-	return pa <= (BIT64(CFG_CORE_ARM64_PA_BITS) - 1);
+	uint8_t b = CFG_CORE_ARM64_PA_BITS;
+
+	if (IS_ENABLED(CFG_AUTO_MAX_PA_BITS)) {
+		const uint8_t map[] = { 32, 36, 40, 42, 44, 48, 52, };
```

Comment:
maybe too much code in an inline function?

Reply:
I agree, it's a bit on the heavy side. Would you prefer to move this into `core/arch/arm/mm/core_mmu.c` or even `core/arch/arm/mm/core_mmu_lpae.c`?

Reply:
`core_mmu_lpae.c`